src/PipelineFramework.LightInject/ServiceRegistryExtensions.cs

```diff
@@ -154,7 +177,16 @@ select new
             //Local functions
             bool IsAsyncPipelineComponent(Type i) => i.IsGenericType && i.GetGenericTypeDefinition() == typeof(IAsyncPipelineComponent<>);
             bool IsPipelineComponent(Type i) => i.IsGenericType && i.GetGenericTypeDefinition() == typeof(IPipelineComponent<>);
-        } 
+        }
+
+        private static Action<IServiceFactory, ProxyDefinition> InterceptorProxyDefinition<TInterceptor>()
+            where TInterceptor : IInterceptor
+            => (factory, proxy) => proxy.Implement(factory.GetInterceptor<TInterceptor>, mi => mi.Name == "Execute");
+
+        private static Action<IServiceFactory, ProxyDefinition> AsyncInterceptorProxyDefinition<TInterceptor>()
+            where TInterceptor : IInterceptor
```

Comment:
Shouldn't this be constrained to AsyncInterceptor instead?

Reply:
Yes, good catch.